packages/fx-core/src/plugins/resource/bot/v2/common.ts

```diff
@@ -0,0 +1,103 @@
+// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT license.
+
+import { Inputs, AzureSolutionSettings } from "@microsoft/teamsfx-api";
+import { Context } from "@microsoft/teamsfx-api/build/v2";
+import { ServiceType } from "../../../../common/azure-hosting/interfaces";
+import { AzureSolutionQuestionNames, BotScenario } from "../../../solution";
+import { QuestionNames, TemplateProjectsConstants, TemplateProjectsScenarios } from "../constants";
+import {
+  AppServiceOptionItem,
+  HostTypeTriggerOptionItem,
+  HostTypeTriggerOptions,
+} from "../question";
+import { HostType, HostTypes, PluginBot } from "../resources/strings";
+import { CodeTemplateInfo } from "./interface/codeTemplateInfo";
+import { langMap, serviceMap, triggerScenarioMap } from "./mapping";
+
+export function getTemplateInfos(ctx: Context, inputs: Inputs): CodeTemplateInfo[] {
+  const lang = resolveProgrammingLanguage(ctx);
+  const scenarios = decideTemplateScenarios(ctx, inputs);
+  return scenarios.map((scenario) => {
+    return {
+      group: TemplateProjectsConstants.GROUP_NAME_BOT,
+      language: lang,
+      scenario: scenario,
+      variables: {},
+    };
+  });
+}
+
+export function decideTemplateScenarios(ctx: Context, inputs: Inputs): string[] {
+  const isM365 = ctx.projectSetting?.isM365;
+  const templateScenarios = [];
+  if (isM365) {
+    templateScenarios.push(TemplateProjectsScenarios.M365_SCENARIO_NAME);
+    return templateScenarios;
+  }
+  const botScenarios = inputs?.[AzureSolutionQuestionNames.Scenarios];
+  if (Array.isArray(botScenarios)) {
+    botScenarios.map((scenario: string) => {
+      switch (scenario) {
+        case BotScenario.CommandAndResponseBot:
+          templateScenarios.push(TemplateProjectsScenarios.COMMAND_AND_RESPONSE_SCENARIO_NAME);
+          break;
+        case BotScenario.NotificationBot:
+          const notificationTriggerType = inputs[QuestionNames.BOT_HOST_TYPE_TRIGGER] as string[];
+          if (notificationTriggerType.includes(AppServiceOptionItem.id)) {
+            templateScenarios.push(TemplateProjectsScenarios.NOTIFICATION_RESTIFY_SCENARIO_NAME);
+          } else {
+            const options = resolveTriggerOption(inputs);
+            templateScenarios.push(
+              TemplateProjectsScenarios.NOTIFICATION_FUNCTION_BASE_SCENARIO_NAME
+            );
+            options.map((option) => templateScenarios.push(triggerScenarioMap[option.trigger!]));
+          }
+          break;
+      }
+    });
+  } else {
```

Comment:
1. botScenarios should be in type of array anyway.
2. The default template case should not be associated with `! isArray`.